src/pty.c

```diff
@@ -462,6 +490,7 @@ static int read_buf(struct kmscon_pty *pty)
 		} else if (errno != EWOULDBLOCK) {
 			log_debug("cannot read from pty of child %d (%d): %m",
 				  pty->child, errno);
+			handle_hup(pty);
```

Comment:
Could you explain this line here? AFAIK, it seems none of the possible errno from read is HUP related, at least according to the man page https://man.archlinux.org/man/core/man-pages/read.2.en

Reply:
When investigating, the first error I've seen is a -EIO reading the pty pipe, when the child process has exited. I then noticed the mask & EV_HUP check, and move the handle_hup() there.
I decided to keep also a call here, in case for some reason kmscon missed the EV_HUP event, but I don't know if it can happen in practice. Anyway this call is harmless, and will do nothing if the child has not exited.

I can either remove this line, or add a comment.

Reply:
Thanks, I've added a comment, and I fixed indentation in handle_hup() while at it.